invoices/pdf_utils.py

```diff
@@ -45,3 +46,27 @@ def generate_hours_pdf_for_invoice(request, invoice):
     # We can generate the pdf from a url, file or, as shown here, a string
     content = render_to_string('pdf_template.html', context=context, request=request)
     return generate_pdf(title, content), title
+
+
+def generate_hours_pdf_for_weekly_report(request, weekly_report):
+    weekly_report_data = get_object_or_404(WeeklyReport, weekly_report_id=weekly_report)
+    title = u"%s - %s - %s-%s" % (weekly_report_data.client,
+                                  weekly_report_data.project,
+                                  weekly_report_data.year,
+                                  weekly_report_data.week)
+    title = title.replace(u"\xe4", u"a").replace(u"\xb6", u"o").replace(u"\x84", u"A").replace(u"\x96", u"O").replace(u"\xf6", "o")
+
+    entries = HourEntry.objects.filter(project=weekly_report_data.project,
+                                       client=weekly_report_data.client,
+                                       date__year__gte=weekly_report_data.year,
+                                       date__week=weekly_report_data.week).filter(incurred_hours__gt=0)
+    phases = {}
```

Comment:
This again is copy-pasted. Do not copy-paste (major) parts when that can be easily avoided.